### ISSUE TITLE
[css-grid-3] `grid-template-areas` in masonry

### ISSUE THREAD

**Issue by Loirooriol** [w3c/csswg-drafts#7995]
https://drafts.csswg.org/css-grid-3/ seems to assume that the explicit grid has 0 tracks in the masonry axis.

This is usually the case, since `grid-template-rows/columns` needs to be set to `masonry` instead of a list of track sizes.
However, `grid-template-areas` can also add tracks into the explicit grid.

I still don't fully understand how masonry works, but testing in Firefox it seems that `grid-template-areas` is allowed to name lines but not create explicit tracks.

**Comment by fantasai** [comment]
Agreed that they should work the same, but that maybe needs clarification in the spec.

**Comment by fantasai** [comment]
I think at this point actually the spec is pretty clear, given the following statements:

> The grid-template-* and grid-auto-rows/grid-auto-columns properties (and their shorthands) apply in the grid axis of the masonry container and establish tracks just as on regular grid containers. (They are ignored in the stacking axis.)

> The explicit grid and implicit grid are formed in the same way as for a regular grid container.

@Loirooriol Do you think anything else needs clarification here?
